### DISCUSSION TITLE
counter  值改变问题

### DISCUSSION THREAD

**Discussion by pengyinghao** [TaleLin/lin-ui#1198] [求助] [answered]
购物车功能 ，商品是一个数组对象，看了下源码，在没有聚焦input的情况下 是直接把input给隐藏的。修改counter值的时候，一旦setdata 就会出现死循环。这种问题该如何处理呢。

**Comment by murongg** [comment, marked as answer]
抱歉，没太明白您想表达的问题，请详细描述，最好也附带代码片段

**Reply by pengyinghao** [reply]
数据结构式这样的
```
[{
      type: 'product',
      children: [{
        id: 1,
        counter: 2
      }, {
        id: 2,
        counter: 3
      }]
    },
    {
      type: 'service',
      children: [{
        id: 3,
        counter: 2
      }, {
        id: 4,
        counter: 3
      }]
}]
```
当我修改某一个对象的counter ,赋值之后需要重新setdata。但是意外的事情发生了,setdata之后 ,counter的input change事件又会重新触发
从而的导致了死循环

![1608608806(1)](https://user-images.githubusercontent.com/34115313/102852208-ac180c00-4458-11eb-928f-34bc496f8039.jpg)

![1608614590(1)](https://user-images.githubusercontent.com/34115313/102852275-dc5faa80-4458-11eb-9ea5-2cd50fe2d73f.jpg)

**Reply by murongg** [reply]
您可以在 `goodsCounterChange` 方法最开始，在 `goodList` 内查找到对应数据，对比 `count` 的值，如改变再去进行相应的 `setData` ，如未改变，则直接 `return`

**Reply by pengyinghao** [reply]
可以给个联系方式嘛 ，我详细给你说一下

**Reply by murongg** [reply]
不好意思哈，不太方便提供联系方式，我们完全可以在此讨论

**Reply by pengyinghao** [reply]
我做了一个代码demo，你看一下 
[demo.zip](https://github.com/TaleLin/lin-ui/files/5728077/demo.zip)

**Reply by murongg** [reply]
代码片段提供请使用：[微信小程序代码片段](https://developers.weixin.qq.com/miniprogram/dev/devtools/minicode.html)

**Reply by pengyinghao** [reply]
https://developers.weixin.qq.com/s/lACLGMmq7omo

**Reply by murongg** [reply]
您可以把这一块
![image](https://user-images.githubusercontent.com/38655613/102856557-ac68d500-4461-11eb-803c-00b6d1273ba3.png)
封装成组件，组件内部负责处理性能问题，会比现在这样处理要简单些

**Reply by pengyinghao** [reply]
但还是会存在 值一改变
 counter又会触发change事件这个问题呀
![1](https://user-images.githubusercontent.com/34115313/102857315-25b4f780-4463-11eb-852e-5d05dcc553a7.gif)

**Reply by murongg** [reply]
您可以去了解一些常用的性能优化相关手段，业务性问题我们无法帮您解决

**Reply by pengyinghao** [reply]
仔细阅读了你说的 ，知道怎么解决了 谢谢了
